### DISCUSSION TITLE
about docker-compose host name?

### DISCUSSION THREAD

**Discussion by MrYZhou** [fastapi/fastapi#6198] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
1.docker-compose like this
version: "3.9"
services:
  mysql:
    restart: always
    image: mysql:5.7

2.connect mysql
DB_HOST = 'mysql'
DB_URL = f"{DB_DRIVER}://{DB_USER}:{DB_PASSWORD}@{DB_HOST}:{DB_PORT}/{DB_NAME}?charset=utf8mb4"
engine = create_engine(DB_URL, echo=SQLMODEL_ECHO)
```

### Description

i use the service name as the host,but it can not work.
according to offical document. it shoud be work.

### Operating System

Linux, Windows

### Operating System Details

win11

### FastAPI Version

0.79

### Python Version

3.10.5

**Comment by jgould22** [comment]
Are you running that code inside a docker container in that docker compose?

Docker service DNS names only resolve to the service from within a container on the same docker network 

Can you link to the part of the documentation you are referring to?

**Comment by JungWooGeon** [comment]
Can you comment on what error occurred?

**Comment by MrYZhou** [comment]
> Are you running that code inside a docker container in that docker compose?
> 
> Docker service DNS names only resolve to the service from within a container on the same docker network
> 
> Can you link to the part of the documentation you are referring to?

Thanks for quiikly reply.

i use docker network inspect command ,and i can see the container mysql and my server are on the same network.
as for error message,it show that can not connect mysql .

raise exc
2022-11-22 15:25:29 sqlalchemy.exc.OperationalError: (pymysql.err.OperationalError) (2003, "Can't connect to MySQL server on 'mysql' ([Errno 111] Connection refused)")
2022-11-22 15:25:29 (Background on this error at: https://sqlalche.me/e/14/e3q8)

i don't know why can the 'mysql' as the host name ? how it work.

**Comment by MrYZhou** [comment]
> Are you running that code inside a docker container in that docker compose?
> 
> Docker service DNS names only resolve to the service from within a container on the same docker network
> 
> Can you link to the part of the documentation you are referring to?

a common error, just show mysql connect error. but if you use computer host 192.168.20.43. it can connect successful. but i need use docker compose services name as the host name.

**Comment by MrYZhou** [comment, marked as answer]
i have resolved, if use docker compose network, this port should use  port inside container
